evm_loader/program/src/entrypoint.rs

```diff
@@ -221,128 +256,338 @@ fn process_instruction<'a>(
                 return Err!(ProgramError::InvalidArgument; "This method should used with Solana sender");
             }
 
-            do_call(program_id, &mut account_storage, accounts, bytes.to_vec(), U256::zero(), u64::MAX)
+            let call_return = do_call(&mut account_storage, accounts, bytes.to_vec(), U256::zero(), u64::MAX)?;
+
+            if let Some(call_results) = call_return {
+                applies_and_invokes(
+                    program_id,
+                    &mut account_storage,
+                    accounts,
+                    None,
+                    call_results)?;
+            }
+
+            Ok(())
         },
-        EvmInstruction::ExecuteTrxFromAccountDataIterative{step_count} =>{
+        EvmInstruction::ExecuteTrxFromAccountDataIterative{collateral_pool_index, step_count} => {
             debug_print!("Execute iterative transaction from account data");
             let holder_info = next_account_info(account_info_iter)?;
             let storage_info = next_account_info(account_info_iter)?;
 
-            let  accounts = &accounts[1..];
+            let operator_sol_info = next_account_info(account_info_iter)?;
+            let collateral_pool_sol_info = next_account_info(account_info_iter)?;
+            let block_acc = next_account_info(account_info_iter)?;
+            let user_eth_info = next_account_info(account_info_iter)?;
+            let system_info = next_account_info(account_info_iter)?;
 
             let holder_data = holder_info.data.borrow();
             let (unsigned_msg, signature) = get_transaction_from_data(&holder_data)?;
 
+            let trx_accounts = &accounts[7..];
+
+            if !operator_sol_info.is_signer {
+                return Err!(ProgramError::InvalidAccountData);
+            }
+
             let trx: UnsignedTransaction = rlp::decode(unsigned_msg).map_err(|e| E!(ProgramError::InvalidInstructionData; "DecoderError={:?}", e))?;
+            let trx_gas_limit = u64::try_from(trx.gas_limit).map_err(|e| E!(ProgramError::InvalidInstructionData; "e={:?}", e))?;
+            let trx_gas_price = u64::try_from(trx.gas_price).map_err(|e| E!(ProgramError::InvalidInstructionData; "e={:?}", e))?;
+            // if trx_gas_price < 1_000_000_000_u64 {
```

Comment:
Since `gas_price` is measured in `wei`, do we need checks here?

Reply:
gas_price is measured in gwei.
1 gwei = 10^9 wei

Reply:
`gas_price` field contains value in `wei`. Even `Metamask` when you set up price in `gwei` sends `wei`.